pkg/publish/default.go

```diff
@@ -94,8 +95,32 @@ func NewDefault(base string, options ...Option) (Interface, error) {
 	return do.Open()
 }
 
+func pushResult(tag name.Tag, br build.Result, opt []remote.Option) error {
+	mt, err := br.MediaType()
+	if err != nil {
+		return err
+	}
+
+	switch mt {
+	case types.OCIImageIndex, types.DockerManifestList:
+		idx, ok := br.(v1.ImageIndex)
+		if !ok {
+			return fmt.Errorf("failed to interpret result as index: %v", br)
+		}
+		return remote.WriteIndex(tag, idx, opt...)
+	case types.OCIManifestSchema1, types.DockerManifestSchema2:
+		img, ok := br.(v1.Image)
+		if !ok {
+			return fmt.Errorf("failed to interpret result as image: %v", br)
+		}
+		return remote.Write(tag, img, opt...)
+	default:
+		return fmt.Errorf("result image media type: %s", mt)
+	}
+}
```

Comment:
I sense a pattern emerging

Reply:
Indeed... could probably do better than this but it's easy enough to understand.

Reply:
Yeah, I wasn't expecting you do change anything here, but we could probably find ways to share the boilerplate if it starts to diverge.